src/mlpack/tests/decision_tree_test.cpp

```diff
@@ -355,6 +557,28 @@ BOOST_AUTO_TEST_CASE(BasicConstructionTest)
 }
 
 /**
+ * Construct a tree with weighted labels.
+ */
+BOOST_AUTO_TEST_CASE(BasicConstructionTestWithWeight)
+{
+  arma::mat dataset(10, 1000, arma::fill::randu);
+  arma::Row<size_t> labels(1000);
+  arma::rowvec weights(labels.n_elem);
+  weights.ones();
+
+  for (size_t i = 0; i < 1000; ++i)
+    labels[i] = i % 3; // 3 classes.
+
+  // Use default parameters.
+  DecisionTree<> wd(dataset, labels, 3, weights, 50);
+  DecisionTree<> d(dataset, labels, 3, 50);
+
+  // Now require that we have some children.
+  BOOST_REQUIRE_GT(wd.NumChildren(), 0);
+  BOOST_REQUIRE_EQUAL(wd.NumChildren(), d.NumChildren());
```

Comment:
You should make the leaf size smaller than 50 so that a full tree is built.  For this case I would say use a leaf size of 1.  Then you will have to adapt the test code here; you can take a look at the Hoeffding tree test to see some code that iterates over all nodes in a tree to ensure that they are the same.  We should definitely end up with the exact same tree as `d` since all the weights are 1.